service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseInitiationController.java

```diff
@@ -49,10 +48,12 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(
 
     @PostMapping("/submitted")
     public void handleSubmittedEvent(
-        @RequestHeader(value = "authorization") String authorization,
-        @RequestHeader(value = "user-id") String userId,
         @RequestBody CallbackRequest callbackRequest) {
 
-        applicationEventPublisher.publishEvent(new InitiatedCaseEvent(callbackRequest, authorization, userId));
+        String caseId = Long.toString(callbackRequest.getCaseDetails().getId());
+        String caseLocalAuthority = (String) callbackRequest.getCaseDetails().getData()
+            .get("caseLocalAuthority");
+
+        localAuthorityUserService.grantUserAccessWithCaseRole(caseId, caseLocalAuthority);
```

Comment:
so yeah what if we did
```
CaseDetails caseDetails = callbackRequest.getCaseDetails;
```

first? Seems to be convention for all our callbacks. 

Very minor, if you don't want to do it I am fine with that :)

Reply:
done :D